src/utils/nps.ts

```diff
@@ -0,0 +1,85 @@
+ *  Licensed under the MIT License. See License.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// Loosely adapted from https://github.com/microsoft/vscode-azure-account/blob/2f497562cab5f3db09f983ab5101040f27dceb70/src/nps.ts
+
+import { env, Memento, MessageItem, Uri, window } from "vscode";
+import { ext } from "vscode-azureappservice/out/src/extensionVariables";
+
+const PROBABILITY = 0.15;
+const MIN_SESSION_COUNT = 10;
+
+const SURVEY_NAME = 'nps1';
+const SURVEY_URL = 'https://www.surveymonkey.com/r/vscodedockernpsinproduct';
+
+const SESSION_COUNT_KEY = `${SURVEY_NAME}/sessioncount`;
+const LAST_SESSION_DATE_KEY = `${SURVEY_NAME}/lastsessiondate`;
+const IS_CANDIDATE_KEY = `${SURVEY_NAME}/iscandidate`;
+
+export async function nps(globalState: Memento): Promise<void> {
+    try {
+        // If not English-language, don't ask
+        if (env.language !== 'en' && !env.language.startsWith('en-')) {
+            return;
+        }
+
+        let isCandidate: boolean | undefined = globalState.get(IS_CANDIDATE_KEY);
+
+        // If not a candidate, don't ask
+        if (isCandidate === false) {
+            return;
+        }
+
+        const date = new Date().toDateString();
+        const lastSessionDate = globalState.get(LAST_SESSION_DATE_KEY, new Date(0).toDateString());
+
+        // If this session is on same date as last session, don't count it
+        if (date === lastSessionDate) {
+            return;
+        }
+
+        // Count this session
+        const sessionCount = globalState.get(SESSION_COUNT_KEY, 0) + 1;
+        await globalState.update(LAST_SESSION_DATE_KEY, date);
+        await globalState.update(SESSION_COUNT_KEY, sessionCount);
+
+        // If under the MIN_SESSION_COUNT, don't ask
+        if (sessionCount < MIN_SESSION_COUNT) {
+            return;
+        }
+
+        // Decide if they are a candidate (if we previously decided they are and they did Remind Me Later, we will not do probability again)
+        // i.e. Probability only comes into play if isCandidate is undefined
+        // tslint:disable-next-line: insecure-random
+        isCandidate = isCandidate || Math.random() < PROBABILITY;
+        await globalState.update(IS_CANDIDATE_KEY, isCandidate);
+
+        // If not a candidate, don't ask
+        if (!isCandidate) {
+            return;
+        }
+
+        const take: MessageItem = { title: 'Take Survey' };
+        const remind: MessageItem = { title: 'Remind Me Later' };
+        const never: MessageItem = { title: 'Don\'t Show Again' };
+
+        // Prompt, treating hitting X as Remind Me Later
+        const result = (await window.showInformationMessage('Do you mind taking a quick feedback survey about the Docker Extension for VS Code?', take, remind, never)) || remind;
+
+        ext.reporter.sendTelemetryEvent('nps', { survey: SURVEY_NAME, response: result.title });

```

Comment:
Should we record the actual display string of the chosen selection or a non-display-string value (so we are free to tweak the display text without complicating telemetry queries)?

Reply:
Yeah, can do that. I had it that way originally but wanted to simplify, down to one line for telemetry.